lib/Controller/MachineTranslationController.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\AppAPI\Controller;
+
+use OCA\AppAPI\AppInfo\Application;
+use OCA\AppAPI\Attribute\AppAPIAuth;
+use OCA\AppAPI\Db\MachineTranslation\MachineTranslationQueueMapper;
+use OCA\AppAPI\Service\ProvidersAI\MachineTranslationService;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\Attribute\NoCSRFRequired;
+use OCP\AppFramework\Http\Attribute\PublicPage;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Http\Response;
+use OCP\AppFramework\OCSController;
+use OCP\DB\Exception;
+use OCP\IConfig;
+use OCP\IRequest;
+
+class MachineTranslationController extends OCSController {
+	protected $request;
+
+	public function __construct(
+		IRequest                                           $request,
+		private readonly IConfig                           $config,
+		private readonly MachineTranslationService         $machineTranslationService,
+		private readonly MachineTranslationQueueMapper     $mapper,
+	) {
+		parent::__construct(Application::APP_ID, $request);
+
+		$this->request = $request;
+	}
+
+	#[NoCSRFRequired]
+	#[PublicPage]
+	#[AppAPIAuth]
+	public function registerProvider(
+		string $name,
+		string $displayName,
+		string $fromLanguages,
```

Comment:
probably "arrays" here?

Reply:
Yeah, missed that, likely we should expect array to encode it later to save in DB.